controllers/storagecluster/cephobjectstores.go

```diff
@@ -175,6 +186,28 @@ func (r *StorageClusterReconciler) newCephObjectStoreInstances(initData *ocsv1.S
 			r.Log.Error(err, "Failed to set ControllerReference for CephObjectStore.", "CephObjectStore", klog.KRef(obj.Namespace, obj.Name))
 			return nil, err
 		}
+		// if kmsConfig is not 'nil', add the KMS details to CephCluster spec
+		if kmsConfigMap != nil {
+			// Set default KMS_PROVIDER and VAULT_SECRET_ENGINE values, refer https://issues.redhat.com/browse/RHSTOR-1963
+			if _, ok := kmsConfigMap.Data["KMS_PROVIDER"]; !ok {
+				kmsConfigMap.Data["KMS_PROVIDER"] = "vault"
+			}
+			kmsConfigMap.Data["VAULT_SECRET_ENGINE"] = "transit"
+			rgwConnDetails := make(map[string]string)
+			for key, value := range kmsConfigMap.Data {
+				// modify or ignore kv engine related options
+				if key == "VAULT_SECRET_ENGINE" {
```

Comment:
can't we just set `rgwConnDetails.Data["VAULT_SECRET_ENGINE"] = "transit"` once the loop is done?

Reply:
that also works